DigitalLearningSolutions.Data/Models/User/UserReference.cs

```diff
@@ -12,5 +12,23 @@ public UserReference(int id, UserType userType)
             Id = id;
             UserType = userType;
         }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is UserReference ur)
+            {
+                return Id == ur.Id && UserType.Equals(ur.UserType);
+            }
+
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                return 93 + 37*(Id.GetHashCode() + 23*UserType.GetHashCode());

```

Comment:
Where are these values from? Is this from the Rider equivalent of this: https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2019


Also, needs a quick formatter run 👁️

Reply:
Ah, it's not, but Rider does do equality members generation and it's better than what I've got, though different from VS' apparently. Do you want me to use Rider's generated code or copy the VS generated code?